src/fsharp/FSharp.Core/fslib-extra-pervasives.fs

```diff
@@ -110,8 +112,34 @@ module ExtraTopLevelOperators =
             member s.GetEnumerator() =
                 // We use an array comprehension here instead of seq {} as otherwise we get incorrect
                 // IEnumerator.Reset() and IEnumerator.Current semantics. 
+                // Coreclr has a bug with SZGenericEnumerators --- implement a correct enumerator.  On desktop use the desktop implementation because it's ngened.
+#if !NETSTANDARD               
                 let kvps = [| for (KeyValue (k,v)) in t -> KeyValuePair (getKey k, v) |] :> seq<_>
                 kvps.GetEnumerator()
+#else
+                let endIndex = t.Count
```

Comment:
I'd prefer this be called `count` since that's how it's used, but this isn't a hill I'm going to die on.

Reply:
@brettfo endindex is the name for that value in the original bcl source code.  I too chuckled over the ridiculousness of the name.